### PR TITLE
[StorageListItem] add definitive StorageList (Order/Wishlist) field to Item

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? |no
| Fixed tickets | #2115 

@TobiSW This is a different approach to the issue. This solves the underlying issue of the relation between order and order-item (eg. storage-list and storage-list-item).

@dkarlovi @solverat @breakone  Any thoughts on that?

### CODE REVIEW

**Comment by solverat** [comment]
Mhmm. It would be interesting to know why the message appears at all. Is it because in V2 the objects were saved more often?

However, it's a relation more but also no more do/while magic.

**Comment by dpfaffenbauer** [comment]
@solverat It happens because the order gets moved and processed before saved. so the items are still in the old location.

**Comment by solverat** [comment]
Ah I see, that's because of the cart/order behavior (same object)!

Well, that's going to be a heavy migration then. :)

**Comment by dpfaffenbauer** [comment]
@solverat exactly. About the migration: Well, yes... There is the other PR https://github.com/coreshop/CoreShop/pull/2116 too, which just does a save before doing the calculations. 

But I think in the long term, it is better to have a real relation between the two.

File: src/CoreShop/Component/Wishlist/Model/Wishlist.php [hunk 1]
```diff
@@ -35,6 +35,13 @@ public function addItem($item): void
     {
         Assert::isInstanceOf($item, WishlistItemInterface::class);
 
+        /**
+         * @var WishlistItemInterface $item
+         */
+        Assert::isInstanceOf($item, WishlistItemInterface::class);
```
Comment (jdreesen) [inline src/CoreShop/Component/Wishlist/Model/Wishlist.php:41]: Now the check is duplicated.

Comment (dpfaffenbauer) [reply src/CoreShop/Component/Wishlist/Model/Wishlist.php]: thanks

**Pull request merged**